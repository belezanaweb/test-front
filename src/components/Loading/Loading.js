import React from 'react'
import { keyframes } from '@emotion/core'
import styled from '@emotion/styled/macro'
import logo from './logo.svg'

const LoadingContainer = styled.div`
  align-items: center;
  background-color: #282c34;
  color: white;
  display: flex;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  justify-content: center;
  min-height: 100vh;
  text-align: center;
`

const logoSpin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`

const LoadingImage = styled.img`
  animation: ${logoSpin} infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
`

function Loading () {
  return (
    <LoadingContainer>
      <LoadingImage src={logo} alt="logo" />
    </LoadingContainer>
  )
}

export default Loading
