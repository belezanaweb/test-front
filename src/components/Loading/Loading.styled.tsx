import styled from 'styled-components/macro'

export const Loading = styled.p`
  font-size: 20px;
`
