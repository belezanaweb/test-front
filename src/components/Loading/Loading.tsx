import React from 'react'

export const Loading = () => <h1>Loading...</h1>
