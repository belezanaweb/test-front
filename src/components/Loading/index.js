import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
import { Container } from './styles';

const useStyles = makeStyles((theme) => ({
    loading: {
        color: '#ffffff',
    },
}));

export default function Loading() {
    const classes = useStyles();

    return (
        <Container>
            <CircularProgress className={classes.loading} />
        </Container>
    );
};