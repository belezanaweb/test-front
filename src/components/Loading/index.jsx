import React from 'react'
import { LoadingBox, Icon } from './styled'
import preloader from './images/preloader.svg'

export default function Loading() {
  return (
    <LoadingBox>
      <Icon src={preloader} alt="Loading" />
    </LoadingBox>
  )
}
