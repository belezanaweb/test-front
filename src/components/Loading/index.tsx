import { Container } from './styles';

export default function Loading() {
  return (
    <Container>
      <div className="lds-ring">
        <div></div>
        <div></div>
        <div></div>
        <div></div>
      </div>
    </Container>
  );
}
