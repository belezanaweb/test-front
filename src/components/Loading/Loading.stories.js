import React from 'react'
import { storiesOf } from '@storybook/react'
import Loading from '.'

storiesOf('components/Loading', module).add('simple', () => <Loading />)
