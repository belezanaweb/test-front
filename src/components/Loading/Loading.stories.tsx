import React from 'react'

import Loading from '.'

export default {
  title: 'Components/Loading',
  component: Loading
}

export const Default: React.FC<{}> = () => <Loading />
