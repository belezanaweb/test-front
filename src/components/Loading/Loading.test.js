import React from 'react'
import { render } from '@testing-library/react'
import Loading from './Loading'

describe('Loading', () => {
  it('should render Loading', () => {
    const { container } = render(<Loading />)

    container.firstChild

    expect(container.firstChild).toBeInTheDocument()
  })
})
