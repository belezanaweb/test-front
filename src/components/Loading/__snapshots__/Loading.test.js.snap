// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render [snapshot] 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #282c34;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 100vh;
  text-align: center;
}

.emotion-0 {
  -webkit-animation: animation-0 infinite 20s linear;
  animation: animation-0 infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

<div>
  <div
    class="emotion-2 emotion-3"
  >
    <img
      alt="logo"
      class="emotion-0 emotion-1"
      src="logo.svg"
    />
  </div>
</div>
`;
